test/PowerShellEditorServices.Test.Host/ServerTestsBase.cs

```diff
@@ -106,20 +111,42 @@ public class ServerTestsBase
             // Start the process
             this.serviceProcess.Start();
 
-            // Wait for the server to finish initializing
-            while (!File.Exists(sessionPath) || (new FileInfo(sessionPath).Length == 0))
+            string sessionDetailsText = string.Empty;
+
+            // Wait up to ~5 seconds for the server to finish initializing
+            var maxRetryAttempts = 10;
+            while (maxRetryAttempts-- > 0)
             {
-                Thread.Sleep(100);
+                try
+                {
+                    using (var stream = new FileStream(sessionPath, FileMode.Open, FileAccess.Read, FileShare.None))
+                    using (var reader = new StreamReader(stream))
+                    {
+                        sessionDetailsText = reader.ReadToEnd();
+                        break;
+                    }
+                }
+                catch (FileNotFoundException)
+                {
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine($"Session details at '{sessionPath}' not available: {ex.Message}");
+                }
+
+                Thread.Sleep(500);
```

Comment:
BTW I timed this while loop in the debugger using VS perf tips and the time to read the file varied between about 350 ms to 1.2 secs so I figured 500 ms was a reasonable sleep time to cut down on the number of retries.